src/pages/Swap/index.tsx

```diff
@@ -194,16 +225,34 @@ export default function Swap({ className }: { className?: string }) {
   // for expert mode
   const [isExpertMode] = useExpertModeManager()
   // swap state
-  const [state, dispatch] = useReducer(swapReducer, initialSwapState)
+  const [state, dispatch] = useReducer(swapReducer, { ...initialSwapState, ...prefilledState })
   const { typedValue, recipient, independentField } = state
+
+  const previousConnectedChainId = usePrevious(connectedChainId)
+  useEffect(() => {
+    const combinedInitialState = { ...initialSwapState, ...prefilledState }
+    console.log({ combinedInitialState })
+    if (previousConnectedChainId && previousConnectedChainId !== connectedChainId) {
```

Comment:
why do the comparison like this instead of just comparing `connectedChainId !== chainId`?

Reply:
i don't think that's the moment we want to clear the inputs. `connectedChainId !== chainId` would be true on the TDP if you're looking at tokens from other chains, and there's no need to run this reset action then (although i think it wouldn't be a noticable change to the user). 

conceptually, we only want to reset the inputs when the connected chain changes, which is what i'm checking here